lib/Backend/Lower.cpp

```diff
@@ -12174,14 +12173,21 @@ Lowerer::GenerateBailOut(IR::Instr * instr, IR::BranchInstr * branchInstr, IR::L
         return bailOutLabel;
     }
 
+    // Add helper label to trigger layout.
+    if (collectRuntimeStatsLabel)
+    {
+        collectRuntimeStatsLabel->Unlink();
```

Comment:
That sounds better to me. The logic in the caller shouldn’t require/expect the label to remain where it was put in the IR stream.

--Paul

From: Rajat Dua [mailto:notifications@github.com]
Sent: Monday, May 9, 2016 5:20 PM
To: Microsoft/ChakraCore <ChakraCore@noreply.github.com>
Cc: Paul Leathers <pleath@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Microsoft/ChakraCore] Fixing bug with shared bailouts (#958)


In lib/Backend/Lower.cpp<https://github.com/Microsoft/ChakraCore/pull/958#discussion_r62595416>:

> @@ -12174,14 +12173,21 @@ Lowerer::GenerateBailOut(IR::Instr * instr, IR::BranchInstr * branchInstr, IR::L

>          return bailOutLabel;

>      }

>

> +    // Add helper label to trigger layout.

> +    if (collectRuntimeStatsLabel)

> +    {

> +        collectRuntimeStatsLabel->Unlink();

Hmm.. The lowering functions for InlineMathFloor et.al. require unlinking now. I could, however, not insert the label in those routines and just pass it to GenerateBailout as collectRuntimeStatsLabel and insert it before the current instruction there.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/958/files/a596859bd23086a43a5027cab91cc8696ac6d331#r62595416>